src/main/java/com/mongodb/jdbc/MongoDatabaseMetaData.java

```diff
@@ -1103,4 +1139,188 @@ public boolean isWrapperFor(Class<?> iface) throws SQLException {
     public <T> T unwrap(Class<T> iface) throws SQLException {
         return (T) this;
     }
+    // Helper functions for getTypeInfo
+    protected Integer getPrecision(String typeName) throws SQLException {
```

Comment:
[organizational nit] My initial vision for this these helper methods was to make them `static`, possibly even part of a totally separate class and `public static`. What if we create a `BsonTypeInfo` class (or some other relevant name) that has all of these as `static` methods? We can also move the `BSON_<type>_TYPE_NAME` constants to that class.

This feels like a big chunk of related fields and methods that could be factored out separately. They are pretty specific to `MongoDatabaseMetaData` subtypes, but still it could be nice to have a larger number of smaller classes than continuing to build these large monolithic classes. What do you think?

Reply:
Okay that makes sense.  I moved the helper functions and the `BSON_<type>_TYPE_NAME` to a new class: `BsonTypeInfo`

Reply:
awesome, thanks